velox/dwio/common/SelectiveFloatingPointColumnReader.h

```diff
@@ -45,7 +45,29 @@ class SelectiveFloatingPointColumnReader : public SelectiveColumnReader {
   readCommon(int64_t offset, const RowSet& rows, const uint64_t* incomingNulls);
 
   void getValues(const RowSet& rows, VectorPtr* result) override {
-    getFlatValues<TData, TRequested>(rows, result, requestedType_);
+    switch (requestedType_->kind()) {
```

Comment:
This method is called per batch, we should not do validation here unless it's not possible else where.  In this case `TData` and `TRequested` is created when we build the column reader, and the validation should be added there instead of this per batch call.

Reply:
@Yuhta Would introducing `TDecode` for internal decoding and limiting `TRequested` to `getValues` be acceptable?

``` c++
template <typename TData, typename TDecode, typename TRequested>
class FloatingPointColumnReader
    : public SelectiveFloatingPointColumnReader<TData, TDecode> {
 public:
  void getValues(const RowSet& rows, VectorPtr* result) override {
    getFlatValues<TData, TRequested>(rows, result, requestedType_);
  }
};
```

 Personally, I prefer keeping the current approach since it aligns with the existing codebase.